src/api/java/com/minecolonies/api/colony/IColonyManager.java

```diff
@@ -371,7 +372,41 @@ static IColonyManager getInstance()
      * @param buildingId ID of the building.
      * @param dim        the dimension.
      */
-    void handleColonyViewRemoveBuildingMessage(int colonyId, BlockPos buildingId, ResourceKey<Level> dim);
+    void handleColonyViewRemoveBuildingMessage(int colonyId, final BlockPos buildingId, final ResourceKey<Level> dim);
+
+    /**
+     * Returns result of {@link IColonyView#handleColonyFieldViewMessage} (BlockPos, FriendlyByteBuf)} if {@link #getColonyView(int, ResourceKey)} gives a not-null result. If {@link
+     * #getColonyView(int, ResourceKey)} is null, returns null.
+     *
+     * @param colonyId ID of the colony.
+     * @param dim      the dimension.
+     * @param type     the field type.
+     * @param position the field position.
+     * @param buf      {@link FriendlyByteBuf} with field data.
+     */
+    void handleColonyFieldViewMessage(
```

Comment:
We try to keep those things out of the colony manager these days. Rather put that in the building manager or so and enough logic in the message itself instead.

Reply:
The client doesn't have a building manager, this is done just like the buildings, where the colony view keeps a map of them.
```java

    @Override
    public IRegisteredStructureManager getBuildingManager()
    {
        return null;
    }
```

Reply:
okay, then just remove it from the colony manager and colony and leave it here. Although we could start thinking of managers for the client side here too to make the class not too big.

Reply:
I will have to keep them for now, as the window actually pulls the field instances from here, so that I have a single point of truth where I know the correct values are.
Alternatively I'd have to rework it back to where the field modules are keeping the instances saved, which I found to not always work as reliably.

Reply:
The colony manager doesnt need it, the colonyview can keep it.